src/llmq/quorums_chainlocks.h

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef DASH_QUORUMS_CHAINLOCKS_H
+#define DASH_QUORUMS_CHAINLOCKS_H
+
+#include "llmq/quorums.h"
+#include "llmq/quorums_signing.h"
+
+#include "net.h"
+#include "chainparams.h"
+
+class CBlockIndex;
+class CScheduler;
+
+namespace llmq
+{
+
+class CChainLockSig
+{
+public:
+    int32_t nHeight{-1};
+    uint256 blockHash;
+    CBLSSignature sig;
+
+public:
+    ADD_SERIALIZE_METHODS
+
+    template<typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(nHeight);
+        READWRITE(blockHash);
+        READWRITE(sig);
+    }
+
+    std::string ToString() const;
+};
+
+class CChainLocksHandler : public CRecoveredSigsListener
+{
+    static const int64_t CLEANUP_INTERVAL = 1000 * 30;
+    static const int64_t CLEANUP_SEEN_TIMEOUT = 24 * 60 * 60 * 1000;
+
+private:
+    CScheduler* scheduler;
+    CCriticalSection cs;
+    bool inInvalidate{false};
```

Comment:
This should probably be `static` or maybe even global atomic and not a member of this class at all. Also, the name is confusing imo, should be smth like `inEnforceBestChainLock` or smth like that. See https://github.com/UdjinM6/dash/commit/3a9c448001a54904aaece792d286abb89618d722.

Reply:
Why global and not a static member of the class?

Reply:
Now that I'm thinking more about it, maybe global is an overkill, agree 🙈